Task 4.2

Mitoschka opened the pull request.

Mitoschka committed:
Task 4.2

Mitoschka committed:
Add Tests

Mitoschka committed:
Update Task 4.2

Mitoschka committed:
Update Task 4.2

yurii-litvinov commented on Task 4.2/Task 4.2.Test/Exception.Test.cs:
```diff
@@ -0,0 +1,38 @@
+{
+    class ExceptionTest
+    {
+        UniqueList unique;
```
private

yurii-litvinov commented on Task 4.2/Task 4.2.Test/List.Test.cs:
```diff
@@ -0,0 +1,33 @@
+        [Test]
+        public void AddUniqueElementToListShouldWorkFalse()
+        {
+            UniqueList list = new UniqueList();
```
Тоже надо вынести в Setup()

yurii-litvinov commented on Task 4.2/Task 4.2/AddException.cs:
```diff
@@ -0,0 +1,18 @@
+        public AddException(string message, Exception inner)
+        : base(message, inner) { }
+        protected AddException(
+        System.Runtime.Serialization.SerializationInfo info,
```
Надо отступы, а то не видно, где заканчивается один метод и начинается другой

yurii-litvinov commented on Task 4.2/Task 4.2/AddException.cs:
```diff
@@ -0,0 +1,18 @@
+namespace MyList
+{
+    [Serializable]
+    public class AddException : Exception
```
Над окомментарии

yurii-litvinov commented on Task 4.2/Task 4.2/MyList.cs:
```diff
@@ -0,0 +1,133 @@
+    /// <param name="position">Number of item to add value to</param>
+    public class MyList
+    {
+        public class ListElement
```
private

yurii-litvinov commented on Task 4.2/Task 4.2/MyList.cs:
```diff
@@ -0,0 +1,133 @@
+        }
+
+        public ListElement head;
+        public int counter;
```
О нет

yurii-litvinov replied:
О нет

yurii-litvinov commented on Task 4.2/Task 4.2/MyList.cs:
```diff
@@ -0,0 +1,133 @@
+        {
+            return counter;
+        }
+    }
```
Хотелось список из 2.1, с операциями добавления/удаления/смены значения по позиции

yurii-litvinov commented on Task 4.2/Task 4.2/UniqueList.cs:
```diff
@@ -0,0 +1,36 @@
+
+namespace MyList
+{
+    public class UniqueList : MyList
```
Надо комментарии

yurii-litvinov replied:
Надо комментарии

yurii-litvinov commented on Task 4.2/Task 4.2/UniqueList.cs:
```diff
@@ -0,0 +1,36 @@
+                ++counter;
+            }
+            throw new AddException("Error: Выражение не являеться уникальным");
+        }
```
Это не поможет выполнить инвариант, что все элементы списка уникальны. Придумайте тест, который не падает и позволяет в UniqueList положить два одинаковых элемента :)

yurii-litvinov replied:
И это не помогло, всё ещё можно иметь UniqueList, элементы которого не уникальны. Подумайте.

yurii-litvinov replied:
И того, что Вы сделали, всё ещё недостаточно

yurii-litvinov replied:
Стало лучше, но список всё ещё может содержать неуникальные элементы

Mitoschka committed:
Add more tests and corrected comments

Mitoschka committed:
Merge branch 'master' into Task-4.2

yurii-litvinov commented on Task 4.2/Task 4.2/DeleteException.cs:
```diff
@@ -0,0 +1,48 @@
+namespace Task_4._2
+{
+    /// <summary>
+    /// Class with implementation exception
```
А чем это отличается от комментария к AddException?

yurii-litvinov commented on Task 4.2/Task 4.2/MyList.cs:
```diff
@@ -0,0 +1,205 @@
+            currentElement.Next.Value = stringOfValue;
+        }
+
+        public bool AddUniqueElementToList(string value, int position)
```
Не, такого метода не должно быть у базового класса

yurii-litvinov commented on Task 4.2/Task 4.2/Program.cs:
```diff
@@ -0,0 +1,25 @@
+        {
+            try
+            {
+                UniqueList list = new UniqueList();
```
var

Mitoschka committed:
Update Task

Mitoschka committed:
Merge branch 'master' into Task-4.2

yurii-litvinov reviewed (changes requested):
Не проходит тест
```csharp
        [Test]
        public void ShouldNotContainDuplicates()
        {
            list.AddElement("abc", 0);
            Assert.Throws<AddException>(() => list.AddElement("abc", 0));
        }
```

Mitoschka committed:
Update Task

1) Added a test test to add a unique item
2) Changed the type of throwing an exception
3) Corrected the name of the method
4) Added the ability to overwrite the method

Mitoschka committed:
Update test of task

fixed a bug in the tests by changing the name of the method

Mitoschka committed:
Add more XML and delete extra libraries

Mitoschka committed:
Merge branch 'master' into Task-4.2

yurii-litvinov commented on Task 4.2/Task 4.2/UniqueList.cs:
```diff
@@ -0,0 +1,65 @@
+        /// </summary>
+        /// <param name="value">value of element</param>
+        /// <param name="position">position of element in list</param>
+        /// <returns></returns>
```
Пустые тэги не нужны

yurii-litvinov commented on Task 4.2/Task 4.2/UniqueList.cs:
```diff
@@ -0,0 +1,65 @@
+        /// <param name="value">value of element</param>
+        /// <param name="position">position of element in list</param>
+        /// <returns></returns>
+        public bool ChecksForUniquenessOfElmenetInList(string value, int position)
```
Зачем оно паблик? И зачем оно принимает position?

Mitoschka committed:
Fix UniqueList and delete wrong metod

Mitoschka committed:
Change XML

Mitoschka committed:
Change Tests, add GetPosition

yurii-litvinov reviewed (changes requested):
Правильный ответ был переопределить SetItemValue. Зачёт не в этот раз, значит

Mitoschka committed:
Fixed error

yurii-litvinov reviewed (approved):
Да, вот теперь всё так. Вам зачёт, который я проставлю на пересдаче.

Mitoschka committed:
Merge branch 'master' into Task-4.2
